src/utils/progress.rs

```diff
@@ -42,6 +42,16 @@ impl ProgressBar {
     }
 
     pub fn finish_with_message(&self, msg: &str) {
+        self.inner.finish_with_message(msg.to_owned());
+        logging::set_progress_bar(None);
+    }
+
+    pub fn finish_with_duration(&self, op: &str) {
+        let e = self.inner.elapsed();
+        let msg = format!("{} completed in {}.{}s", op, e.as_secs(), e.as_millis());
+        let progress_style = ProgressStyle::default_bar().template("{prefix:.dim} {msg}");
+        self.inner.set_style(progress_style);
+        self.inner.set_prefix(">");
         self.inner.finish_with_message(msg);
         logging::set_progress_bar(None);
```

Comment:
Shouldn't we use `self.finish_with_message(&msg)` to avoid duplicating the logic here?

Reply:
I was thinking about it but decided no to do that to keep everything in line. This way, every method is a standalone thing that just proxies data to `indicatif` methods. It's very rarely touched part of code and adds 1 more line of code anyway.